src/plone/app/multilingual/tests/test_selector.py

```diff
@@ -930,3 +931,146 @@ def assertUrlsEqual(self, url1, url2):
+        self.portal_url = self.portal.absolute_url()
+        self.request = self.layer['request']
+        alsoProvides(self.layer['request'], IPloneAppMultilingualInstalled)
+
+        # Setup testbrowser
+        self.browser = Browser(self.layer['app'])
+        self.browser.handleErrors = False
+        registry = getUtility(IRegistry)
+        self.settings = registry.forInterface(ILanguageSchema, prefix="plone")
+
+    def test_set_language_is_present(self):
+        """ test the presence of set_language parameter in the urls created in the language selector"""
+
+        self.settings.set_cookie_always = False
+        self.selector_viewlet = LanguageSelectorViewlet(
+            self.portal['en'], self.request, None, None
+        )
+        self.selector_viewlet.update()
+
+        selector_languages = self.selector_viewlet.languages()
+        tg = ITG(self.portal['en'])
+
+        self.assertEqual(
+            selector_languages,
+            [
+                {
+                    'code': u'en',
+                    u'flag': u'/++resource++country-flags/gb.gif',
+                    u'name': u'English',
+                    u'native': u'English',
+                    'url': SELECTOR_VIEW_TEMPLATE
+                    % {
+                        'url': self.portal.absolute_url(),
+                        'tg': tg,
+                        'lang': 'en',
+                        'query': '?set_language=en',
+                    },
+                    'selected': True,
+                    'translated': True,
+                },
+                {
+                    'code': u'ca',
+                    u'flag': u'/++resource++language-flags/ca.gif',
+                    u'name': u'Catalan',
+                    u'native': u'Catal\xe0',
+                    'url': SELECTOR_VIEW_TEMPLATE
+                    % {
+                        'url': self.portal.absolute_url(),
+                        'tg': tg,
+                        'lang': 'ca',
+                        'query': '?set_language=ca',
+                    },
+                    'selected': False,
+                    'translated': True,
+                },
+                {
+                    'code': u'es',
+                    u'flag': u'/++resource++country-flags/es.gif',
+                    u'name': u'Spanish',
+                    u'native': u'Espa\xf1ol',
+                    'url': SELECTOR_VIEW_TEMPLATE
+                    % {
+                        'url': self.portal.absolute_url(),
+                        'tg': tg,
+                        'lang': 'es',
+                        'query': '?set_language=es',
+                    },
+                    'selected': False,
+                    'translated': True,
+                },
+            ],
+        )
+
+
+    def test_set_language_is_not_present_when_always_set_cookie_is_set(self):
+        """ test the presence of set_language parameter in the urls created in the language selector"""
```

Comment:
Should be `...test the absence of set_language...`

Reply:
indeed... I copied over the method and did not change the comment :face_with_head_bandage: